File: src/PokeData.Domain/Regions/Region.cs
using PokeData.Domain.Generations;

namespace PokeData.Domain.Regions;

public class Region
{
  public byte Number { get; set; }

  public string UniqueName { get; set; } = string.Empty;
  public string? DisplayName { get; set; }

  public Generation? MainGeneration { get; set; }
}

File: frontend/src/types/region.ts
import type { Aggregate } from "./aggregate";
import type { Generation } from "./generation";
import type { SearchPayload, SortOption } from "./search";

export type Region = Aggregate & {
  number: number;
  uniqueName: string;
  displayName?: string;
  mainGeneration?: Generation;
};

export type RegionSort = "DisplayName" | "Number" | "UniqueName" | "UpdatedOn";

export type RegionSortOption = SortOption & {
  field: RegionSort;
};

export type SearchRegionsPayload = SearchPayload & {
  numberIn: number[];
  sort: RegionSortOption[];
};
